mido/midifiles/midifiles.py

```diff
@@ -399,18 +399,19 @@ def play(self, meta_messages=False):
 
         You will receive copies of the original messages, so you can
         safely modify them without ruining the tracks.
-
         """
-        sleep = time.sleep
-        now = time.time
-        t0 = now()
-        file_time = 0
+        t0 = time.time()
+        combined_time = 0.0
 
         for msg in self:
-            file_time += msg.time
-            duration_to_next_event = file_time - (now() - t0)
-            if (duration_to_next_event > 0):
-                sleep(duration_to_next_event)
+            combined_time += msg.time
+
+            # we compute the difference between the nominal combined time
+            # and the actual time spent at this point
+            duration_to_next_event = combined_time - (time.time() - t0)
+
+            if (duration_to_next_event > 0.0):
```

Comment:
I understand use of `0.0` for the variable initialization, since it's a cheap form of type annotation.  However in this expression no variable is being annotated, and it's perfectly valid to compare a float to an int, so `0` is the nominal form.

Reply:
I have no strong opinion on this, but I think `0.0` helps a little bit extra here since it makes it clear that we're dealing with seconds and not MIDI ticks (which would always be integers).

Reply:
@bast, can you remove the parentheses from around the expression?

    if duration_to_next_event > 0.0:

Reply:
Will do - I was programming something else in Fortran at the same time and it polluted me :-)